dotenv-linter/src/checks/schema_validator.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashSet;
+
+use super::Check;
+use crate::{
+    common::{LintKind, Warning},
+    schema::SchemaValueType,
+};
+use dotenv_lookup::LineEntry;
+use email_address::EmailAddress;
+use url::Url;
+pub(crate) struct SchemaChecker<'a> {
+    schema: Option<&'a crate::schema::DotEnvSchema>,
+    seen_keys: HashSet<String>,
+    last_line_number: usize,
+}
+
+impl<'a> SchemaChecker<'a> {
+    pub fn new(opts: &'a crate::cli::options::CliOptions) -> Self {
+        Self {
+            schema: opts.schema.as_ref(),
+            seen_keys: HashSet::new(),
+            last_line_number: 0,
+        }
+    }
+}
+
+impl Check for SchemaChecker<'_> {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        self.schema?;
+        self.last_line_number = line.number;
+        let schema = self.schema.unwrap();
```

Comment:
```suggestion
        let schema = self.schema?;
        self.last_line_number = line.number;
       
```

Reply:
lol - i originally has `if let Some()......` clippy said , replace with this i just pasted it in without thinking. fixed